pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -0,0 +1,134 @@
+
+
+# Inserting Custom JavaScript 
+
+AMP pages support custom JavaScript through the `<amp-script>` component. 
+
+
+```html
+<!doctype html>
+<html ⚡>
+<head>
+  ...
+  <script async custom-element="amp-script" src="https://cdn.ampproject.org/v0/amp-script-0.1.js"></script>
+<body>  
+  ...
+  <amp-script layout="container" src="https://example.com/myfile.js">
+    <p>Initial content that can be modified from JavaScript</p>
+  </amp-script>
+  ...
+</body>
+</html>
+
+```
+
+
+The `<amp-script>` component registers a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) to run on a separate thread than the main page. The Web Worker is given its own copy of the DOM through `amp-script` use of [Worker DOM](https://github.com/ampproject/worker-dom). This allows the Web Worker to use JavaScript libraries, such as [React](https://reactjs.org/) and [jQuery](https://jquery.com/), without modification. 
+
+The `amp-script` component sends messages between the Web Worker thread and the main thread, causing any changes the user makes on the main DOM to be echoed on the Web Worker's false DOM. In turn, the Web Worker can then update the false DOM, which is reflected on the main DOM.
+
+[tip type="important"]
+ `<amp-script>` is in experimental mode and may break in unpredictable ways.
+[/tip]
+
+## Custom Scripts Caching 
+
+The [AMP cache](https://www.ampproject.org/docs/fundamentals/how_cached) serves custom JavaScript files inserted with `<amp-script>` same as AMP component scripts. This ensures that any custom JavaScript will not slow the speed of an AMP document. 
+
+The AMP cache proxies the JavaScript files and then delivers them. Users can expect the same performance experience from a page using `<amp-script>` as a page that doesn't include it. 
+
+# Limitations to `<amp-script>`
+
+To keep experiences on AMP pages consistent, limitations exist on `<amp-script>` to guarantee a fast loading time and smooth UI. 
+
+## Initialization
+
+JavaScript inside the Web Worker allows minimal change to the DOM on load. Changes allowed during this phase are:
+
+
+*   Registering event handlers.
+*   Splitting a TextNode into multiple TextNodes to allow for styling liberties, assuming the actual text doesn't change.
+
+The DOM inside `<amp-script>` tags should be almost identical before and after initialization. 
+
+
+## Interaction
+
+To ensure a responsive UX, DOM manipulations in response to a user interaction must happen in **less than one second**.  If a script mutates the DOM in response to a user interactions after one second, the `amp-script` component will terminate the Web Worker. A terminated `<amp-script>` component will not run again.
```

Comment:
There's two new rules here:

1. If the script runs the `fetch` function as a result of user interaction, the period of time to manipulate the DOM is extended from 1 second to 5 seconds.

2. If the `<amp-script>` uses a fixed layout (anything other than `layout=container` *and* has a height smaller than 300, anything is allowed, there's no user interaction requirement. The "300" is subject to change, so let's write "sufficiently small" instead until a decision is made.